### PR TITLE
refactor: Notifications

### LINKED ISSUE #185: Reduce lines of code for showing ElNotification
Right now showing an ElNotification on frontend needs around 6 lines of code

[Example:](https://github.com/R2NorthstarTools/FlightCore/blob/main/src-vue/src/views/DeveloperView.vue#L115-L120)
```js
ElNotification({
    title: 'Success',
    message: "Disabled all mods but core",
    type: 'success',
    position: 'bottom-right'
});
```

Now some are unavoidable but at the very least the `position: 'bottom-right'` is used by every notification we send. Maybe there's a way to set the defaults so that we can at least remove that line?

### PR DESCRIPTION
**Pull request by Alystrasz**
Wraps up notification calls in dedicated methods.

Closes #185.

```typescript
/**
 * Displays content to the user in the form of a notification appearing on screen bottom right.
 **/
function showNotification(
    title: string,
    message: string = '',
    type: 'success' | 'warning' | 'error' | 'info' = 'success',
    duration: number = 4500
): NotificationHandle;

/**
 * Helper method displaying an error message to the user.
 **/
function showErrorNotification(error: string): NotificationHandle;
```

### CODE REVIEW

File: src-vue/src/views/mods/LocalModsView.vue [hunk 1]
```diff
@@ -85,12 +85,11 @@ export default defineComponent({
...
-                    title: this.$t('generic.error'),
-                    message: `${error}`,
-                    type: 'error',
-                    position: 'bottom-right'
-                });
+                showNotification(
+                    this.$t('generic.error'),
+                    `${error}`,
+                    'error'
+                );
```
Comment (GeckoEidechse) [inline src-vue/src/views/mods/LocalModsView.vue:92]: Is there a reason this isn't a `showErrorNotification`? ^^

File: src-vue/src/plugins/store.ts [hunk 2]
```diff
@@ -18,6 +18,8 @@ import { NorthstarMod } from "../../../src-tauri/bindings/NorthstarMod";
 import { searchModule } from './modules/search';
 import { i18n } from '../main';
 import { pullRequestModule } from './modules/pull_requests';
+import { PullsApiResponseElement } from "../../../src-tauri/bindings/PullsApiResponseElement";
+import {showErrorNotification, showNotification} from '../utils/ui';
```
Comment (GeckoEidechse) [inline src-vue/src/plugins/store.ts:22]: Formatting :P
```suggestion
import { showErrorNotification, showNotification } from '../utils/ui';
```

File: src-vue/src/plugins/store.ts [hunk 3]
```diff
@@ -18,6 +18,8 @@ import { NorthstarMod } from "../../../src-tauri/bindings/NorthstarMod";
 import { searchModule } from './modules/search';
 import { i18n } from '../main';
 import { pullRequestModule } from './modules/pull_requests';
+import { PullsApiResponseElement } from "../../../src-tauri/bindings/PullsApiResponseElement";
```
Comment (GeckoEidechse) [inline src-vue/src/plugins/store.ts:21]: Is this import used anywhere in the file? At least I don't see it ^^

File: src-vue/src/components/ThunderstoreModCard.vue [hunk 4]
```diff
@@ -72,9 +72,9 @@ import {invoke, shell} from "@tauri-apps/api";
 import {ThunderstoreModStatus} from "../utils/thunderstore/ThunderstoreModStatus";
 import {NorthstarMod} from "../../../src-tauri/bindings/NorthstarMod";
 import {GameInstall} from "../utils/GameInstall";
-import {ElNotification} from "element-plus";
 import { NorthstarState } from "../utils/NorthstarState";
 import { ElMessageBox } from "element-plus";
+import {showErrorNotification, showNotification} from "../utils/ui";
```
Comment (GeckoEidechse) [inline src-vue/src/components/ThunderstoreModCard.vue:77]: Formatting
```suggestion
import { showErrorNotification, showNotification } from "../utils/ui";
```

File: src-vue/src/plugins/modules/pull_requests.ts [hunk 5]
```diff
@@ -1,8 +1,8 @@
-import { ElNotification } from "element-plus";
 import { invoke, shell } from "@tauri-apps/api";
 import { PullsApiResponseElement } from "../../../../src-tauri/bindings/PullsApiResponseElement";
 import { PullRequestType } from '../../../../src-tauri/bindings/PullRequestType';
 import { store } from "../store";
+import {showErrorNotification, showNotification} from "../../utils/ui";
```
Comment (GeckoEidechse) [inline src-vue/src/plugins/modules/pull_requests.ts:5]: Formatting
```suggestion
import { showErrorNotification, showNotification } from "../../utils/ui";
```

File: src-vue/src/utils/ui.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+        position: 'bottom-right',
+    });
+}
+
+/**
+ * Helper method displaying an error message to the user.
+ **/
+function showErrorNotification(error: string): NotificationHandle {
+    return showNotification(i18n.global.tc('generic.error'), error, 'error');
+}
```
Comment (GeckoEidechse) [inline src-vue/src/utils/ui.ts:24]: Does TypeScript have optional function arguments like e.g. Python?

Cause then we could give `title` a default value of `i18n.global.tc('generic.error')` and if an error message has a special title, then we set it explicitly just for that.

File: src-vue/src/views/SettingsView.vue [hunk 7]
```diff
@@ -72,10 +72,10 @@
 
 <script lang="ts">
 import { defineComponent } from "vue";
-import { ElNotification } from 'element-plus';
 import { invoke } from "@tauri-apps/api";
 import { ReleaseCanal } from "../utils/ReleaseCanal";
 import { Store } from 'tauri-plugin-store-api';
+import {showErrorNotification, showNotification } from "../utils/ui";
```
Comment (GeckoEidechse) [inline src-vue/src/views/SettingsView.vue:78]: Formatting
```suggestion
import { showErrorNotification, showNotification } from "../utils/ui";
```

File: src-vue/src/views/RepairView.vue [hunk 8]
```diff
@@ -78,33 +68,22 @@ export default defineComponent({
...
-            const notification = ElNotification({
-                title: 'Force reinstalling Northstar',
-                message: 'Please wait',
-                duration: 0,
-                type: 'info',
-                position: 'bottom-right'
-            });
+            const notification = showNotification(
+                'Force reinstalling Northstar',
+                'Please wait',
```
Comment (GeckoEidechse) [inline src-vue/src/views/RepairView.vue:73]: Looks like we forgot to localise that. I'll make an issue to address it in a separate PR ^^

File: src-vue/src/views/RepairView.vue [hunk 9]
```diff
@@ -32,11 +32,11 @@
 
 <script lang="ts">
 import { defineComponent } from "vue";
-import { ElNotification } from "element-plus";
 import { GameInstall } from "../utils/GameInstall";
 import { invoke } from "@tauri-apps/api";
 import { ReleaseCanal } from "../utils/ReleaseCanal";
 import { Store } from 'tauri-plugin-store-api';
+import {showErrorNotification, showNotification} from "../utils/ui";
```
Comment (GeckoEidechse) [inline src-vue/src/views/RepairView.vue:39]: Formatting
```suggestion
import { showErrorNotification, showNotification } from "../utils/ui";
```

File: src-vue/src/views/DeveloperView.vue [hunk 10]
```diff
@@ -87,10 +87,10 @@
 <script lang="ts">
 import { defineComponent } from "vue";
 import { invoke } from "@tauri-apps/api";
-import { ElNotification } from "element-plus";
 import { GameInstall } from "../utils/GameInstall";
 import { TagWrapper } from "../../../src-tauri/bindings/TagWrapper";
 import PullRequestsSelector from "../components/PullRequestsSelector.vue";
+import {showErrorNotification, showNotification } from "../utils/ui";
```
Comment (GeckoEidechse) [inline src-vue/src/views/DeveloperView.vue:93]: Formatting
```suggestion
import { showErrorNotification, showNotification } from "../utils/ui";
```

**Review by GeckoEidechse — COMMENTED** [review]
Aight, initial review done. Thanks a lot for the refactor <3

File: src-vue/src/utils/ui.ts (hunk 6 above)
Comment (Alystrasz) [reply src-vue/src/utils/ui.ts:24]: We could do that, but we should put `title` argument after the `error` one; if you want to see a working example, check `showNotification` prototype:
```typescript
function showNotification(
    title: string,
    message: string = '',
    type: 'success' | 'warning' | 'error' | 'info' = 'success',
    duration: number = 4500
)
```

Comment (GeckoEidechse) [reply src-vue/src/utils/ui.ts:24]: Would be fine by me ^^

Comment (Alystrasz) [reply src-vue/src/utils/ui.ts]: Addressed in f41241b3a612f6807d229f7f125505bece495eda

File: src-vue/src/views/mods/LocalModsView.vue [hunk 11]
```diff
@@ -35,10 +35,10 @@
 
 <script lang="ts">
 import { invoke } from '@tauri-apps/api';
-import { ElNotification } from 'element-plus';
 import { defineComponent } from 'vue';
 import { GameInstall } from '../../utils/GameInstall';
 import { NorthstarMod } from "../../../../src-tauri/bindings/NorthstarMod";
+import {showErrorNotification, showNotification} from '../../utils/ui';
```
Comment (GeckoEidechse) [inline src-vue/src/views/mods/LocalModsView.vue:41]: Formatting >:)

```suggestion
import { showErrorNotification, showNotification } from '../../utils/ui';
```

**Review by GeckoEidechse — COMMENTED** [review]
Checcking code changes, looks all good except for one formatting issue :P

Doing testing soon™

**Review by GeckoEidechse — APPROVED** [review]
Testing also went fine on Linux rn. So once the formatting issue is resolved it would be ready to merge from my side ^^

**Pull request merged**